src/cmap/conn/stream_description.rs

```diff
@@ -1,10 +1,12 @@
+use std::time::Duration;
+
 use crate::{is_master::IsMasterReply, sdam::ServerType};
 
 /// Contains information about a given server in a format digestible by a connection.
-#[derive(Debug, Default)]
+#[derive(Debug, Default, Clone)]
 pub(crate) struct StreamDescription {
-    /// The type of the server.
-    pub(crate) server_type: ServerType,
+    /// The type of the server when the handshake occurred.
+    pub(crate) initial_server_type: ServerType,
```

Comment:
>I think it is actually possible for the server type to have changed without the connections having closed

This is definitely true. The "connections survive primary stepdown" tests verify it.

>This does make me wonder if using the StreamDescription for building the operation could lead issues, though, since in that case, an operation to the former primary might not send $readPreference due to the StreamDescription assuming that it's a primary still. This is unrelated to the changes in this PR though, so no need to do anything about it here.

Agreed. I know we've discussed `StreamDescription` many times, but I think I still don't exactly know why we use it over a `ServerDescription` in operation execution.